TodoApp.ConsoleApp/UI/UserInput.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+
+namespace TodoApp.ConsoleApp.UI
+{
+    public static class UserInput
+    {
+        const string Yes = "yes";
+        const string No = "no";
+
+        private static string ReadInput(bool required = false, string defaultValue = "")
+        {
+            string text = "";
+            if (required)
+            {
+                text = Console.ReadLine();
+
+                while (string.IsNullOrEmpty(text))
+                {
+                    Console.WriteLine("This field can't be empty!");
+                    text = Console.ReadLine();
+                }
+            }
+
+            else
+            {
+                text = Console.ReadLine();
+
+                if (string.IsNullOrEmpty(text))
+                {
+                    text = defaultValue;
+                }
+            }
+
+            return text;
+        }
+
+        public static string ReadText(string question, bool required = false, string defaultValue = "")
```

Comment:
This method is wrong. It is not doing what it is intended.
The implementation should be just:
```csharp
Console.WriteLine(question);
string text = ReadInput(required, defaultValue);
return text;
```

`ReadInput()` is checking the `required` flag and setting the `defaultValue` when needed. You don't need to do this here also. That's why we added the `required` and `defaultValue` arguments there.

Reply:
Do i need to make another method to, print message, because this way if i use ReadText as im used it now, if i try to print any message ReadInput will be called after that.

Reply:
So here is what we can do: merge `ReadInput()` and `ReadText()` into single method `ReadText(question, required, defaultValue)`. Use this method in the other Read methods.

```csharp
string ReadText(string question = "", bool required = false, string defaultValue = "")
{
    if (!string.IsNullOrEmpty(question)
    {
         Console.Write($"{question} ");
    }

    string text = Console.ReadLine();
    if (required)
    {
        while (string.IsNullOrEmpty(text))
        {
            Console.WriteLine("This field can't be empty!");
            text = Console.ReadLine();
        }
    }
    else if (string.IsNullOrEmpty(text))
    {
        text = defaultValue;
    }

    return text;
}
```

Reply:
In `ReadOption()` you can use `ReadText("Command:", required, defaultValue.ToString())`
In `ReadYesNo()` -  `ReadText(question, required, defaultValue.ToString())`